features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/api.js

```diff
@@ -829,7 +829,7 @@ class API extends Resource {
      * @param api {Object} Updated API object(JSON) which needs to be updated
      */
     update(updatedProperties) {
-        const updatedAPI = { ...this.toJSON(), ...updatedProperties };
+        const updatedAPI = { ...this.toJSON(), ...this.toJSON(updatedProperties) };
```

Comment:
@tharikaGitHub  I think we don't need to pass the updated content to `toJson`, Calling api.update without valid API content is not recommended